### ISSUE TITLE
Catch exceptions in std::stod

### ISSUE THREAD

**Issue by lscalfi** [chemfiles/cfiles#20]
In utils.cpp, we should handle invalid arguments (for steps and unitcell parsers) using a try{}catch(invalid_argument){}.
For now, in case of wrong argument (eg --cell=sdt:yr:4) the program gives: "Error stod"

**Comment by Luthaf** [comment]
(FYI, you can use single backticks \` for showing code: `code`, and triple backticks \`\`\` for multiline code. See the [markdown](https://guides.github.com/features/mastering-markdown/) reference about this)

The most maintainable way to do this would be to have a separated function for string => double conversions, and use it everywhere.

```cpp
double string2double(const std::string& string) {
    try {
       return std::stod(string);
    } catch (const std::invalid_argument&) {
        throw CFilesError("Can not convert '" + string + "' to number");
    }
}
```

**Comment by Luthaf** [comment]
This was fixed in #22.
